### ISSUE TITLE
Emphasis wrapping of highlighted text

### ISSUE THREAD

**Issue by porglezomp** [demon386/SmartMarkdown#35]
This is a feature request based on a feature in MarkdownEditing (which I can't use reliably for a few reasons.)
`*` and `_` should behave like parenthesis and quotes do in other code, when you select some text and type `*`, it should wrap that text with the `*`. When there is a second paired `*` in your text, typing another should simply skip it (like typing a closing quote when one has been provided.)

Basically, match the behavior of Sublime's quotes.
